rust/fastsim-core/src/traits.rs

```diff
@@ -19,9 +19,14 @@ pub trait SerdeAPI: Serialize + for<'a> Deserialize<'a> {
     /// # Returns:
     ///
     /// A Rust Result
-    fn to_file(&self, filename: &str) -> Result<(), anyhow::Error> {
+    fn to_file(&self, filename: &str) -> anyhow::Result<()> {
         let file = PathBuf::from(filename);
-        match file.extension().unwrap().to_str().unwrap() {
+        match file
+            .extension()
+            .ok_or_else(|| anyhow!("Unable to parse file extension: {:?}", file))?
+            .to_str()
+            .unwrap()
```

Comment:
@kylecarow , I'd lean toward also getting rid of this unwrap with a separate message since it's not much additional effort.

Reply:
@calbaker done, see latest commit